extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -14,29 +17,61 @@ import {
  */
 let ShoppingConfigResponseDef;
 
+/**
+ * Uses the specified validation configuration to run validation against
+ * the user's shopping configuration.
+ * @param {!ShoppingConfigDataDef} productConfig The user's config object.
+ * @return {boolean} returns a boolean indicating whether the validation
+ *     was successful.
+ */
+function validateConfig(productConfig) {
+  let isValidConfig = true;
+  const errors = validateProduct(productConfig);
+
+  if (errors.length > 0) {
+    isValidConfig = false;
+    user().warn('AMP-STORY-SHOPPING-CONFIG', `${errors}`);
+  }
+
+  return isValidConfig;
+}
```

Comment:
- Take a `name` string to identify each product, the `validateProduct` function can take it.
- nit: short-circuit return.

```suggestion
function validateConfig(productConfig, name) {
  const errors = validateProduct(productConfig, name);
  if (errors.length > 0) {
    user().warn('AMP-STORY-SHOPPING-CONFIG', `${errors}`);
    return false;
  }
  return true;
}
```